modules/javafx.graphics/src/main/java/javafx/scene/image/WritableImage.java

```diff
@@ -145,17 +172,33 @@ boolean pixelsReadable() {
         return true;
     }
 
+    void bufferDirty(Rectangle rect) {
+        getWritablePlatformImage().bufferDirty(rect);
+        pixelsDirty();
+    }
+
+    private static PixelBuffer validatePixelBuffer(PixelBuffer pixelBuffer) {
+        Objects.requireNonNull(pixelBuffer, "PixelBuffer must be specified.");
+        return pixelBuffer;
+    }
+
     private PixelWriter writer;
     /**
      * This method returns a {@code PixelWriter} that provides access to
-     * write the pixels of the image.
+     * write the pixels of the image. {@code PixelWriter} is not supported for
+     * images constructed from a {@link PixelBuffer}.
      *
      * @return the {@code PixelWriter} for writing pixels to the image
+     *
+     * @throws UnsupportedOperationException if this image is created using a {@link PixelBuffer}
      */
     public final PixelWriter getPixelWriter() {
         if (getProgress() < 1.0 || isError()) {
             return null;
         }
+        if (pixelBuffer != null) {
+            throw new UnsupportedOperationException("PixelWriter is not supported with PixelBuffer");
+        }
```

Comment:
I would move this check before the progress check.

Reply:
moved the check